src/main/java/com/asu/EduMentor/model/DBConnection.java

```diff
@@ -0,0 +1,48 @@
+package com.asu.EduMentor.model;
+
+import lombok.Getter;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public final class DBConnection {
+    private static volatile DBConnection instance;
```

Comment:
what does volatile do?

Reply:
Volatile here is used to ensure that all threads see if any update happens to the shared resource(the DB instance in our code).It also ensures the full construction of the instance before it is visible to other threads so that we avoid hard to diagnose bugs(prevents the JVM to reorder the instructions related to this variable).